CONTRIBUTING.md

```
@@ -0,0 +1,86 @@
+- `types.rs` contains core data structures and error types
+
+Key patterns:
+
+- Section weights: add new section weights in `container/*.rs` using existing match patterns. Higher weight means more likely to contain useful strings.
+- Semantic tags: add new Tag variants in `types.rs`, implement detection in `classification/semantic.rs`, and update any tag merging logic if needed.
+- Deduplication: preserve all occurrences and merge tags across occurrences in `extraction/dedup.rs`.
+- Public structs: keep public API structs non_exhaustive and provide explicit constructors.
+- Imports: prefer `stringy::extraction` or `stringy::types`. Do not import locally-defined types inside `extraction/mod.rs`.
+
+## Tests
+
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)